### PR TITLE
Tests for linear vesting

### PR DESCRIPTION
**Pull request by mpetruska**
* [x] refactor
* [x] test for happy path
* [x] ready for review

### CODE REVIEW

File: src/LinearVesting.hs [hunk 1]
```diff
@@ -1,19 +1,57 @@
...
+  , vestingPeriodEnd :: Integer
+  , firstUnlockPossibleAfter :: Integer
+  , totalInstallments :: Integer
+  }
+  deriving stock (Show)
+
+PlutusTx.makeLift ''VestingDatum
+PlutusTx.makeIsDataIndexed ''VestingDatum [('VestingDatum, 0)]
+
+data PAssetClassData (s :: S) = PAssetClassData (Term s (PDataRecord '["symbol" ':= PCurrencySymbol, "tokenName" ':= PTokenName]))
```
Comment (mpetruska) [inline src/LinearVesting.hs:42]: Note: if the type with the same name is used from Plutarch.Extra, the datum deserialization fails with `unListData (Const 0 [...])`.

File: test/Spec.hs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module Main (main) where
+
+import Test.Tasty (defaultMain, testGroup)
+
+import LinearVestingSpec qualified
+
+main :: IO ()
+main =
+  defaultMain $ testGroup "smart-handles" [LinearVestingSpec.tests]
```
Comment (solidsnakedev) [inline test/Spec.hs:9]: smart-handles -> vesting

File: src/LinearVesting.hs (hunk 1 above)
Comment (solidsnakedev) [reply src/LinearVesting.hs:42]: no derivation in Plutarch.Extra.AssetClass?

Comment (mpetruska) [reply src/LinearVesting.hs:42]: So, there are a couple, the one that we can use here is the data encoded one `PAssetClassData`. But that cannot be deserialized at runtime, due to the error I mentioned above.

File: test/Spec.hs (hunk 2 above)
Comment (mpetruska) [reply test/Spec.hs]: Fixed, thank you.

**Pull request merged**